service/server.py

```diff
@@ -67,21 +68,38 @@ def get_properties_for_address(address):
     return query.execute().hits
 
 
+def paginated_address_records(address_records, page_number):
+    start_index = (page_number-1)*int(SEARCH_RESULTS_PER_PAGE)
+    end_index = max(page_number*int(SEARCH_RESULTS_PER_PAGE), len(address_records)-1)
```

Comment:
If the intention is to make sure we don't go beyond the array boundaries, then we should replace `max` with `min`.

Also, the second argument should be `len(address_records)` (without `-1` at the end), as the item from `end_index` wouldn't be included in the resulting subarray.

I also might have gone crazy, so please check me before making the change.